src/prefect/_internal/compatibility/experimental.py

```diff
@@ -0,0 +1,106 @@
+"""
+Utilities for experimental features.
+
+Experimental features have a group, feature name, and optional help string.
+
+When an experimental feature is used, a warning will be displayed. Warnings may be
+disabled per feature group with the setting `PREFECT_EXPERIMENTAL_WARN_<GROUP>`. 
+Warnings may also be disabled globally with the setting `PREFECT_EXPERIMENTAL_WARN`.
+
+Some experimental features require opt-in to enable any usage. These require the setting
+`PREFECT_EXPERIMENTAL_ENABLE_<GROUP>` to be set or an error will be thrown on use.
+"""
+import functools
+import warnings
+from typing import Callable, TypeVar
+
+from prefect.settings import PREFECT_EXPERIMENTAL_WARN, SETTING_VARIABLES, Setting
+
+T = TypeVar("T", bound=Callable)
+
+EXPERIMENTAL_WARNING = (
+    "{feature} is experimental and the interface or behavior may change without warning. "
+    "We recommend pinning to a specific version to avoid breaking changes. "
+    "{help} "
+    "To disable warnings for this group, disable PREFECT_EXPERIMENTAL_WARN_{group}. "
+)
+
+EXPERIMENTAL_ERROR = (
+    "{feature} is experimental and requires opt-in for usage. "
+    "{help} "
+    "To use this feature, enable PREFECT_EXPERIMENTAL_ENABLE_{group}. "
+)
```

Comment:
Nit: I forget, do line breaks come through with `Warning` messages?  If so, I think we should break these over lines so they are a little more clean.